config.go

```diff
@@ -354,53 +379,76 @@ func loadConfig() (*config, error) {
 	cfg.LitecoindMode.Dir = cleanAndExpandPath(cfg.LitecoindMode.Dir)
+	cfg.Tor.V2PrivateKeyPath = cleanAndExpandPath(cfg.Tor.V2PrivateKeyPath)
 
-	// Setup dial and DNS resolution functions depending on the specified
-	// options. The default is to use the standard golang "net" package
-	// functions. When Tor's proxy is specified, the dial function is set to
-	// the proxy specific dial function and the DNS resolution functions use
-	// Tor.
-	cfg.net = &torsvc.RegularNet{}
-	if cfg.Tor.Socks != "" && cfg.Tor.DNS != "" {
-		// Validate Tor port number
-		torport, err := strconv.Atoi(cfg.Tor.Socks)
-		if err != nil || torport < 1024 || torport > 65535 {
-			str := "%s: The tor socks5 port must be between 1024 and 65535"
-			err := fmt.Errorf(str, funcName)
-			fmt.Fprintln(os.Stderr, err)
-			fmt.Fprintln(os.Stderr, usageMessage)
-			return nil, err
-		}
+	// Ensure that the user didn't attempt to specify negative values for
+	// any of the autopilot params.
+	if cfg.Autopilot.MaxChannels < 0 {
+		str := "%s: autopilot.maxchannels must be non-negative"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		return nil, err
+	}
+	if cfg.Autopilot.Allocation < 0 {
+		str := "%s: autopilot.allocation must be non-negative"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		return nil, err
+	}
+	if cfg.Autopilot.MinChannelSize < 0 {
+		str := "%s: autopilot.minchansize must be non-negative"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		return nil, err
+	}
+	if cfg.Autopilot.MaxChannelSize < 0 {
+		str := "%s: autopilot.maxchansize must be non-negative"
+		err := fmt.Errorf(str, funcName)
+		fmt.Fprintln(os.Stderr, err)
+		return nil, err
+	}
 
-		// If ExternalIPs is set, throw an error since we cannot
-		// listen for incoming connections via Tor's SOCKS5 proxy.
-		if len(cfg.ExternalIPs) != 0 {
-			str := "%s: Cannot set externalip flag with proxy flag - " +
-				"cannot listen for incoming connections via Tor's " +
-				"socks5 proxy"
-			err := fmt.Errorf(str, funcName)
-			return nil, err
-		}
+	// Ensure that the specified values for the min and max channel size
+	// don't are within the bounds of the normal chan size constraints.
+	if cfg.Autopilot.MinChannelSize < int64(minChanFundingSize) {
+		cfg.Autopilot.MinChannelSize = int64(minChanFundingSize)
+	}
+	if cfg.Autopilot.MaxChannelSize > int64(maxFundingAmount) {
+		cfg.Autopilot.MaxChannelSize = int64(maxFundingAmount)
+	}
 
-		cfg.net = &torsvc.TorProxyNet{
-			TorDNS:          cfg.Tor.DNS,
-			TorSocks:        cfg.Tor.Socks,
+	// Validate the Tor config parameters.
+	cfg.Tor.SOCKS = normalizeAddress(
+		cfg.Tor.SOCKS, strconv.Itoa(defaultTorSOCKSPort),
+	)
+	cfg.Tor.DNS = normalizeAddress(
+		cfg.Tor.DNS, strconv.Itoa(defaultTorDNSPort),
+	)
+	cfg.Tor.Control = normalizeAddress(
+		cfg.Tor.Control, strconv.Itoa(defaultTorControlPort),
+	)
+	switch {
+	case !cfg.Tor.V2 && !cfg.Tor.V3:
```

Comment:
At this point, if either of them are active, we should ensure that `lnd` is _only_ listening on local host.

Reply:
IMO it shouldn't be assumed that if the user wants to use tor for outbound connections, then they _also_ want to use it for inbound connections. So if v2 and v3 aren't set, then we should also disable listening for the p2p interface all together.